### DISCUSSION TITLE
Run a notebook file and export to HTML or PDF

### DISCUSSION THREAD

**Discussion by eford** [fonsp/Pluto.jl#1149] [Questions about using Pluto] [answered]
I'd like a way to suggest an option to run a pluto notebook as a script and have it export the resulting notebook to an HTML or PDF file.

**Comment by fonsp** [comment, marked as answer]
You can use [PlutoSliderServer.jl](https://github.com/JuliaPluto/PlutoSliderServer.jl):
```julia
julia> using PlutoSliderServer

julia> PlutoSliderServer.export_notebook("./Tower of Hanoi.jl")
```

**Reply by eford** [reply]
Thanks.  This has worked well for my class. 

Is there a good way to rerun a template notebook but with some variables changes (like the PlutoRESTClient server does) AND export the changed notebook to html (or even a standard notebook?). export_notebook doesn't seem to with with a PlutoNotebook from the REST client.

Thanks.

**Reply by fonsp** [reply]
Cool question! Can you make a new discussion about this?

**Reply by eford** [reply]
Now in [#1705](https://github.com/fonsp/Pluto.jl/discussions/1705)

**Comment by dpo** [comment]
The answer shows how to export to html, but how to export to pdf from the command line?

**Reply by fonsp** [reply]
Hey! You can use PlutoPDF.jl :


```julia
import PlutoPDF
PlutoPDF.pluto_to_pdf("notebook.jl")
```
